list.h

```diff
@@ -1,97 +1,282 @@
- *   each implementation must create its own.
- *     - For example, a singly linked list might have `Node` defined as
- *       `private` inside its class, as a `struct` containing a `Node * next`
- *       and an `int value`.  A doubly linked list would also need its `Node`
- *       to contain a `Node * previous`.
- */
+* Copyright &copy; 2015 Duy Nguyen <andy21996@csu.fullerton.edu>
+* Released under the [MIT License] (http://opensource.org/licenses/MIT)
+* ------------------------------------------------------------------------- */
 
+#include "stdafx.h"
+#include "list1.h"
+#include <iostream>
+using std::cout;
+using std::endl;
+
+int main()
+{
+	List1 * l1 = new List1;
+
+	// Test insert function
+	for (int i = 1; i <= 10; i++)
+	{
+		l1->insert(i, i * 10);
+	}
+
+	l1->print();
+
+	// Test peek function
+	for (int i = 1; i <= 10; i++)
+		cout << "Value of Node #" << i << " is: " << l1->peek(i) << endl; 
+
+	// Test remove function for invalid Node
+	if (l1->remove(15) == 0)
+		cout << "15 is out of index range \n";
+
+	// Test remove function for last Node
+	if (l1->remove(10) > 0)
+	{
+		cout << "The list after the Node #10 was removed: \n";
+		l1->print();
+	}
+
+	// Test remove function for middle Node
+	if (l1->remove(3) > 0)
+	{
+		cout << "The list after the Node #3 was removed: \n";
+		l1->print();
+	}
+
+	// Test remove function for first Node
+	if (l1->remove(1))
+	{
+		cout << "The list after the Node #1 was removed: \n";
+		l1->print();
+	}
+	return 0;
+}
+-------------------------------------------------------------------
+List1.h
+#ifndef LIST1_H
+#define LIST1_H
+
+#include "list.h"
+
+class Node
+{
+public:
+	Node();
+	Node(const Node& nextNode, int myValue);
+	~Node();
+	Node * getNext() const;
+	int getValue() const;
+	void setNext(Node * nextNode);
+	void setValue(int newValue);
+
+private:
+	Node * next;
+	int value;
+};
+
+class List1 : public List
+{
+public:
+	List1();
+	~List1();
+	int length() const;
+	void insert(int index, const int value);
+	int remove(int index);
+	int peek(int index) const;
+	Node * getmHead() const;
+	void print() const;
+
+private:
+	Node * mHead;
+	int mLength;
+};
+
+#endif
+-------------------------------------------------------------------
+List1.cpp
+#include "stdafx.h"
+#include <iostream>
+#include "list1.h"
+using std::cout;
+using std::endl;
+
+Node::Node()
+{
+	next = NULL;
+	value = 0;
+}
+Node::Node(const Node& nextNode, int myValue)
+{
+	*next = nextNode;
+	value = myValue;
+}
+Node::~Node() {}
+Node *Node::getNext() const
+{
+	return next;
+}
+int Node::getValue() const
+{
+	return value;
+}
+void Node::setNext(Node * nextNode)
+{
+	next = nextNode;
+}
+void Node::setValue(int newValue)
+{
+	value = newValue;
+}
+
+List1::List1()
+{
+	mHead = NULL;
+	mLength = 0;
+}
+List1::~List1() {}
+int List1::length() const
+{
+	return mLength;
+}
+void List1::insert(int index, const int value)
+{
+	if (index <= 0 || index > mLength + 1)
+		cout << "Error \n";
```

Comment:
this is a valid way to do it, and perhaps more clear than normalizing the index (as i did).  as a note: if i were writing this, i would probably put a `return` inside this `if` statement, rather than putting the rest of the method body inside an `else` block -- but there are situations (e.g. when writing code for a military drone -- they have crazy strict standards for code that runs on those things) where this would be the way you'd have to do it.